addons/web/static/src/js/views/search_panel_model_extension.js

```diff
@@ -0,0 +1,616 @@
+odoo.define("web/static/src/js/views/search_panel_model_extension.js", function (require) {
+    "use strict";
+
+    const ActionModel = require("web/static/src/js/views/action_model.js");
+    const { deepEqual, sortBy } = require("web.utils");
+    const Domain = require("web.Domain");
+    const pyUtils = require("web.py_utils");
+
+    // DefaultViewTypes is the list of view types for which the searchpanel is
+    // present by default (if not explicitly stated in the 'view_types' attribute
+    // in the arch).
+    const defaultViewTypes = ["kanban", "tree"];
+    let nextSectionId = 1;
+
+    const SEARCH_PANEL_DEFAULT_LIMIT = 200;
+
+    const hasValues = (s) => {
+        if (s.errorMsg) {
+            return true;
+        } else if (s.groups) {
+            return [...s.groups.values()].some((g) => g.values.size);
+        } else if (s.type === 'category') {
+            return s.values && s.values.size > 1; // __all__ item ignored
+        } else {
+            return s.values && s.values.size > 0;
+        }
+    };
+    const isObject = (obj) => typeof obj === "object" && obj !== null;
+
+    class SearchPanelModelExtension extends ActionModel.Extension {
+        constructor() {
+            super(...arguments);
+
+            this.searchDomain = null;
+            this.DEFAULT_VALUE_INDEX = 0;
+        }
+
+        get(property, ...args) {
+            switch (property) {
+                case "domain": return this.getDomain();
+                case "sections": return this.getSections(...args);
+            }
+        }
+
+        async reloadAfterDispatch() {
+            await this._fetchSections();
+        }
+
+        async load() {
+            await this._fetchSections({ initialLoad: true });
+        }
+
+        prepareState() {
+            Object.assign(this.state, { sections: new Map(), });
+            this._createSectionsFromArch();
+            if (this.config.defaultNoFilter) {
+                const categories = this.categories;
+                for (const { fieldName } of categories) {
+                    this.config.defaultValues[fieldName] =
+                        this.config.defaultValues[fieldName] || false;
+                }
+            }
+        }
+
+        //---------------------------------------------------------------------
+        // Actions / Getters
+        //---------------------------------------------------------------------
+
+        getDomain() {
```

Comment:
inline doc should be reviewed in this file. Some logic has changed too and some comments might have become
inaccurate

Reply:
Done

Reply:
Done